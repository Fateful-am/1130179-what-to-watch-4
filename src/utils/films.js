export const MovieInfoMode = {
  CARD: `card`,
  PROMO: `promo`,
  OVERVIEW: `overview`,
  DETAIL: `detail`,
  REVIEWS: `reviews`
};

export const getMovies = (sourceMovies, sliceFrom, count) => {
  return sourceMovies.slice(sliceFrom, sliceFrom + count).map((movie) => {
    return {
      id: movie.id,
      title: movie.title,
      posterImage: movie.posterImage,
    };
  });
};

export const getMoviePromoMode = (sourceMovies, index) => {
  const movie = sourceMovies[index];
  return {
    id: movie.id,
    title: movie.title,
    genre: movie.genre,
    posterImage: movie.posterImage,
    backgroundImage: movie.backgroundImage,
    released: movie.released
  };
};
