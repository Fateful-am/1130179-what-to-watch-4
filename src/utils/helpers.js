import moment from 'moment';

export const convertToImageName = (movieName) => {
  return movieName.toLowerCase().split(` `).join(`-`).split(`:`).join(``);
};

export const getMovieById = (movies, id) => {
  const movieIndex = movies.findIndex((movie) => movie.id === id);
  if (movieIndex > -1) {
    return movies[movieIndex];
  }
  return null;
};

export const formatDuration = (duration) => {
  if (duration < 60) {
    return moment.utc(moment.duration(duration, `minutes`).asMilliseconds()).format(`m[m]`);
  }
  return moment.utc(moment.duration(duration, `minutes`).asMilliseconds()).format(`H[h] m[m]`);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};


export const smallMovieCards = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`
];
