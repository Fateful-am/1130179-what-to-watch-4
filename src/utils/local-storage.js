import {extend} from './helpers';

const ItemsType = {
  AUTH_STATUS: `authStatus`,
  LAST_URL: `lastUrl`,
};

export default class LocalStorage {
  constructor(key, storage) {
    this._storage = storage;
    this._storeKey = key;
  }

  _getItems() {
    try {
      return JSON.parse(this._storage.getItem(this._storeKey)) || {};
    } catch (err) {
      return {};
    }
  }

  _getItem(item) {
    const store = this._getItems();
    return store[item];
  }

  _setItem(item, value) {
    const store = this._getItems();
    this._storage.setItem(
        this._storeKey,
        JSON.stringify(extend(store, {
          [item]: value,
        }))
    );
  }

  getAuthStatus() {
    return this._getItem(ItemsType.AUTH_STATUS);
  }

  setAuthStatus(authStatus) {
    this._setItem(ItemsType.AUTH_STATUS, authStatus);
  }

  getLastUrl() {
    return this._getItem(ItemsType.LAST_URL) || `/`;
  }

  setLastUrl(url) {
    this._setItem(ItemsType.LAST_URL, url);
  }
}
