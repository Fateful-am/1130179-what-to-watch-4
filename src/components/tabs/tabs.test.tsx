import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Tabs from './tabs';
import {noop} from '../../utils/helpers';

describe(`Tabs component for MoviePage render correctly:`, () => {
  const mockClassNames = {
    list: `movie-nav__list`,
    item: `movie-nav__item`,
    activeItem: `movie-nav__item--active`,
    link: `movie-nav__link`
  };

  it(`with Overview tab`, () => {
    const tree = renderer
    .create(
        <Tabs
          tabs={[`Overview`, `Details`, `Reviews`]}
          className={mockClassNames}
          activeTab={`Overview`}
          onTabClick={noop}
        />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with Details tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`Overview`, `Details`, `Reviews`]}
            className={mockClassNames}
            activeTab={`Details`}
            onTabClick={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with Reviews tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`Overview`, `Details`, `Reviews`]}
            className={mockClassNames}
            activeTab={`Reviews`}
            onTabClick={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe(`Tabs component for Main render correctly:`, () => {
  const mockClassNames = {
    list: `catalog__genres-list`,
    item: `catalog__genres-item`,
    activeItem: `catalog__genres-item--active`,
    link: `catalog__genres-link`
  };

  it(`with "All genres" tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`All genres`, `Comedy`, `Drama`]}
            className={mockClassNames}
            activeTab={`All genres`}
            onTabClick={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with "Comedy" tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`All genres`, `Comedy`, `Drama`]}
            className={mockClassNames}
            activeTab={`Comedy`}
            onTabClick={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with "Drama" tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`All genres`, `Comedy`, `Drama`]}
            className={mockClassNames}
            activeTab={`Drama`}
            onTabClick={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
