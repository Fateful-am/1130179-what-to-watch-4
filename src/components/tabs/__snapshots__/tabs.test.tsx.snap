// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component for Main render correctly: with "All genres" tab 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
</ul>
`;

exports[`Tabs component for Main render correctly: with "Comedy" tab 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
</ul>
`;

exports[`Tabs component for Main render correctly: with "Drama" tab 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
</ul>
`;

exports[`Tabs component for MoviePage render correctly: with Details tab 1`] = `
<ul
  className="movie-nav__list"
>
  <li
    className="movie-nav__item"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item movie-nav__item--active"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Reviews
    </a>
  </li>
</ul>
`;

exports[`Tabs component for MoviePage render correctly: with Overview tab 1`] = `
<ul
  className="movie-nav__list"
>
  <li
    className="movie-nav__item movie-nav__item--active"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Reviews
    </a>
  </li>
</ul>
`;

exports[`Tabs component for MoviePage render correctly: with Reviews tab 1`] = `
<ul
  className="movie-nav__list"
>
  <li
    className="movie-nav__item"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item movie-nav__item--active"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Reviews
    </a>
  </li>
</ul>
`;
